pkg/subctl/cmd/deploybroker.go

```diff
@@ -83,7 +83,7 @@ var deployBroker = &cobra.Command{
 		status := cli.NewStatus()
 		status.Start("Deploying broker")
 		err = broker.Ensure(config)
-		status.End(err == nil)
+		status.End(cli.Success)
```

Comment:
Sorry, I didn’t make this clear. Whenever the original code has a condition on `err`, you need to preserve that, with the conversion function you had in the previous iteration. So `status.end(true)` becomes `status.end(cli.Success)`, `status.end(false)` becomes `status.end(cli.Failure)`, and `status.end(err == nil)` becomes `status.end(cli.CheckStatus(err))`. The idea is to use `err` as an indicator of success: if, when we call `status.End()`, `err` is `nil`, that indicates that the process succeeded, otherwise it failed.

Reply:
Isn't 'err == nil' true ? Also status.end doesn't do anything functionally except for notifying status to print message according to status.QueueSuccessMessage or status.QueueFailureMessage. Even if I am calling CheckStatus(err) what would it return? true or false?? Which again translates to success or failure.

Reply:
`err == nil` is true if `err` is `nil`, false otherwise. The idea here is that if `broker.Ensure(config)` succeeds, it will return no error, and `status.End(true)` is called to indicate success; if it fails, it will return an error, and `status.End(false)` is called to indicate failure.

`CheckStatus(err)` should return `cli.Success` is `err` is `nil`, `cli.Failure` otherwise.

Reply:
Put another way,

    status.End(err == nil)

is equivalent to

    if (err == nil) {
        status.End(true)
    } else {
        status.End(false)
    }

Reply:
Got it

Reply:
Got it. THanks :)